### ISSUE TITLE
Support for storage_account blocks in azurerm_linux_function_app & azurerm_windows_function_app

### ISSUE THREAD

**Issue by blueelvis** [hashicorp/terraform-provider-azurerm#18688]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to mount a file share in my azurerm_windows_function_app  & `azurerm_linux_function_app` resource. This functionality seems available in `azurerm_windows_web_app` & `azurerm_linux_web_app`. I have accomplished adding a file share to my function app using the azure-cli's az webapp config storage-account add command and pointing it to my function app, but the terraform provider seems to be missing this functionality.

### New or Affected Resource(s)/Data Source(s)

azurerm_linux_function_app, azurerm_windows_function_app 

### Potential Terraform Configuration

```hcl
resource "azurerm_linux_web_app" "example" {
  name                = "example"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_service_plan.example.location
  service_plan_id     = azurerm_service_plan.example.id

  site_config {}
# Potential feature
  storage_account {
    name = "MyShare"
    type = "AzureFiles"
    account_name = azurerm_storage_account.my_other_account.name
    share_name = "share-in-the-account"
    access_key = azurerm_storage_account.my_other_account.access_key
    mount_path = "/some/directory"
  }
}

resource "azurerm_windows_web_app" "example" {
  name                = "example"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_service_plan.example.location
  service_plan_id     = azurerm_service_plan.example.id

  site_config {}
# Potential feature
  storage_account {
    name = "MyShare"
    type = "AzureFiles"
    account_name = azurerm_storage_account.my_other_account.name
    share_name = "share-in-the-account"
    access_key = azurerm_storage_account.my_other_account.access_key
    mount_path = "/some/directory"
  }
}
```

### References

I am not sure why but there was a similar issue which was closed without any comment - https://github.com/hashicorp/terraform-provider-azurerm/issues/6790

Following link describes the availability of this feature - 
1. https://learn.microsoft.com/en-us/azure/azure-functions/scripts/functions-cli-mount-files-storage-linux
2. https://learn.microsoft.com/en-us/azure/app-service/configure-connect-to-azure-storage?tabs=portal&pivots=code-windows - Although this feature is described for Windows Web App, I have tried the same steps with a Windows Function App on Consumption Tier & and it worked. The only piece is that there needs to be documentation added as in the argument only accepts, `/mounts/blah-blah` and then the mount is going to be available in `C:\mounts\blah-blah`

**Comment by xiaxyi** [comment]
Thanks @blueelvis , if my understanding is correct, the mount file share in function app is supported in TF, the setting is being handled in a different way. Have you tried to set `WEBSITE_RUN_FROM_PACKAGE ` key in `app_setting` as mentioned in this [docs](https://learn.microsoft.com/en-us/azure/azure-functions/run-functions-from-deployment-package)

Feel free to let me know if I'm taking it in a wrong way~

**Comment by blueelvis** [comment]
> Thanks @blueelvis , if my understanding is correct, the mount file share in function app is supported in TF, the setting is being handled in a different way. Have you tried to set `WEBSITE_RUN_FROM_PACKAGE ` key in `app_setting` as mentioned in this [docs](https://learn.microsoft.com/en-us/azure/azure-functions/run-functions-from-deployment-package)
> 
> Feel free to let me know if I'm taking it in a wrong way~

@xiaxyi - That is indeed a different thing. What I am talking about is having functionality to mount an Azure Storage File Share to the Function App. It is not related to the WEBSITE_RUN_FROM_PACKAGE setting which dictates where the code is stored for the Functions. Consider this like a NFS mount to the Function app.

More information on the use case over here - https://learn.microsoft.com/en-us/azure/azure-functions/scripts/functions-cli-mount-files-storage-linux


If you don't mind, I would like to work on this. I have been looking at the `windows_web_app_resource.go`  & `linux_web_app_resource.go`  and all I need to do is implement (Basically copy-paste) the `StorageAccounts` related config and setup and then paste it in `windows_function_app_resource.go`  & `linux_function_app_resource.go` .

Could you please guide me if I am thinking this correct or not?

**Comment by blueelvis** [comment]
Fixed by #18760
